src/main/scala/play/api/libs/mailer/MailerPlugin.scala

```diff
@@ -273,3 +281,40 @@ case class AttachmentData(name: String,
                           mimetype: String,
                           description: Option[String] = None,
                           disposition: Option[String] = None) extends Attachment
+
+case class SMTPConfiguration(host: String,
+                             port: Int,
+                             ssl: Boolean = false,
+                             tls: Boolean = false,
+                             user: Option[String],
+                             password: Option[String],
+                             debugMode: Boolean = false,
+                             timeout: Option[Int] = None,
+                             connectionTimeout: Option[Int] = None)
+
+object SMTPConfiguration {
```

Comment:
@jroper is there a better way to build SMTPConfiguration from both `PlayConfig` and `Configuration` ?

I'm using `Configuration` because `getOptional` method throws an exception if the value is undefined. But when we configure programmatically the mailer instance I don't want to throw an exception because we have default values.

With PlayConfig the following code will throw an exception because `user` is undefined:

```scala
mailer.configure(Configuration.from(Map("host" -> "typesafe.org", "port" -> 1234))).send(email)
``` 

With Configuration `user` will be resolved to `None` and the mailer client will have no authentication.